src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -154,6 +155,16 @@ public BiomeEntry(Biome biome, int weight)
         }
     }
 
+    public static BiomeEntry getWeightedBiomeEntry(BiomeType type, INoiseRandom context)
+    {
+        List<BiomeEntry> biomeList = biomes[type.ordinal()];
+
+        int totalWeight = WeightedRandom.getTotalWeight(biomeList);
+        int weight = isTypeListModded(type) ? context.random(totalWeight) : context.random(totalWeight / 10) * 10;
+
+        return WeightedRandom.getRandomItem(biomeList, weight);
+    }
```

Comment:
This method can probably be renamed to `getWeightedBiome` and just return the `Biome`, since the weight value is only needed here and not in BiomeLayer.

Reply:
True, but it may be useful for other things that we haven't thought about to use this method and get the weight

Reply:
Even if the weight value is useful for other processes, those other processes should probably be handled within `BiomeManager` or in other processes called from it, and not done within `BiomeLayer`, which would also require further patching it. At the point that this method returns, it should only be the `Biome`, since that it all that is consequential in `BiomeLayer`.

Reply:
@Xerousso You can leave it like this.
It does not hurt to have it return `BiomeEntry` instead of biome.